library/spdm_responder_lib/libspdm_rsp_common.c

```diff
@@ -24,12 +24,11 @@ uint16_t libspdm_allocate_rsp_session_id(const libspdm_context_t *spdm_context)
     return (INVALID_SESSION_ID & 0xFFFF0000) >> 16;
 }
 
-libspdm_return_t
-libspdm_build_opaque_data_version_selection_data(const libspdm_context_t *spdm_context,
-                                                 size_t *data_out_size,
-                                                 void *data_out)
+void libspdm_build_opaque_data_version_selection_data(const libspdm_context_t *spdm_context,
+                                                      size_t *data_out_size,
+                                                      void *data_out)
 {
-    size_t final_data_size;
+    size_t final_data_size = libspdm_get_opaque_data_version_selection_data_size(spdm_context);
```

Comment:
I don't recommend we do assignment in definition.

Can we move it just before ASSERT?

Reply:
Will do.